Source/WebCore/dom/Element.cpp

```diff
@@ -4740,6 +4742,52 @@ bool Element::isSpellCheckingEnabled() const
     return true;
 }
 
+bool Element::isWritingSuggestionsEnabled() const
+{
+    // If none of the following conditions are true, then return `false`.
+
+    // `element` is an `input` element whose `type` attribute is in either the
+    // `Text`, `Search`, `URL`, `Email` state and is `mutable`.
+    auto isEligibleInputElement = [&] {
+        RefPtr input = dynamicDowncast<HTMLInputElement>(this);
+        if (!input)
+            return false;
+
+        auto isAllowedField = input->isTextField() || input->isSearchField() || input->isURLField() || input->isEmailField();
```

Comment:
`isTextField()` encompasses all of these and more. I don't think it's what you want.

We probably want InputType's textTypes although that includes Telephone. Not including Telephone seems like a specification issue?